root/static/scripts/edit/externalLinks.js

```diff
@@ -1603,6 +1610,10 @@ function isGoogleAmp(url) {
   return /^https?:\/\/([^/]+\.)?google\.[^/]+\/amp/.test(url);
 }
 
+function isGoogleSearch(url) {
+  return /^https?:\/\/([^/]+\.)?google\.[^/]+\/search/.test(url);
```

Comment:
This wouldn't match `https://link.example?ref=http://google.com/search?q=foo` as it is (notice it expects a dot at the end of whatever is before Google) but it would match something like `https://link.example?ref=www.google.com/search?q=foo`. While that doesn't seem too likely to happen, we might as well use your version to be safe :)